src/core/src/runtime/string_aligned_buffer.cpp

```diff
@@ -82,22 +82,25 @@ namespace ov {
 StringAlignedBuffer::StringAlignedBuffer(size_t num_elements, size_t byte_size, size_t alignment, bool initialize)
     : AlignedBuffer(byte_size, alignment),
       m_num_elements(num_elements) {
-    OPENVINO_ASSERT(sizeof(std::string) * num_elements <= byte_size + alignment,
-                    "Allocated memory of size " + std::to_string(byte_size) + " bytes is not enough to store " +
-                        std::to_string(num_elements) + " std::string objects");
+    const auto has_enough_size = (sizeof(std::string) * num_elements) <= (byte_size + alignment);
+    OPENVINO_ASSERT(has_enough_size,
```

Comment:
As I understand, AlignedBuffer allocates char array inside (char *) and ctor of AlignedBuffer is already called when this exception occurs. I expect that the destructor of AlignedBuffer won't be called in this case. It looks like a memory leak.
Should we deallocate memory manually or use smart pointers?

Reply:
The `AlignedBuffer` is created here and have dtor. After exception it will be called and release allocated memory. The `StringAlignedBuffer` is not created yet and its dtor will be not called. 
Other created members will have destructor and will be used. There should not be memory leak.

Regarding smart pointer I assume that `AlignedBuffer` was intentionally designed to not use smart pointers.

Reply:
valgrind actually showed me 1 "possibly lost"  block but now I see this is not related to this allocation, it seems to be related to the exception itself , 144 bytes always. Probably not a real issue.
double checked, you are right, dtor will was called.

Reply:
I've an issue with check of `StringAlignedBuffer` size in test `create_not_initialized_but_init_before_destruction` which found on debug build using MSVC some kind of "heap corruption". Maybe this was reported by valgrind.